### PR TITLE
fixing the functionalities in getItems and searchProduct tasks -#184651344-#184651352

### PR DESCRIPTION
**Pull request by JulesNIYOMWUNGERI**
#### What does this PR do?
This PR is for fixing some functionalities in getItems and searchProduct tasks
#### Description of Task to be completed?

- In getItems task when there is no products on the page requested, It will return status of 404, and message describes it.
- In searchProduct task when requested product is not found, it will return status of 404, and message describes it 

#### What are the relevant pivotal tracker/Trello stories?

- #184651344
- #184651352

### CODE REVIEW

**Review by bahati10 — DISMISSED** [review]
Looks Good

**Review by kagame8732 — DISMISSED** [review]
Now get Item and search product working property.

**Review by nshutip — DISMISSED** [review]
Receiving the variables from the request body is a better practice.

File: src/controllers/searchProducts.controller.js [hunk 1]
```diff
@@ -4,35 +4,39 @@ import db from '../Database/models';
...
-  if (name) {
-    where.name = { [Op.iLike]: `%${name}%` };
-  }
-
-  if (category) {
-    where.category = { [Op.iLike]: `%${category}%` };
+  try {
+    const {
+      name, category, minPrice, maxPrice
+    } = req.query;
```
Comment (didasmbalanya) [inline src/controllers/searchProducts.controller.js]: use a validation middleware to check for this values

**Review by didasmbalanya — DISMISSED** [review]

**Review by ruberwa — APPROVED** [review]

**Review by kagame8732 — APPROVED** [review]
Well implemented!

**Pull request merged**